### PR TITLE
[Android] Added support for MicrosoftBotIdentifier

### PR DESCRIPTION
**Pull request by AikoBB**
# Description

Introducing a new type of CommunicationIdentifier - MicrosoftBotIdentifier and the logic is covered with tests.
Implementation of the [User Story 3133111](https://skype.visualstudio.com/SPOOL/_workitems/edit/3133111): [SDK][Common][Mobile] As a developer I can identify if a user is a Teams User or Bot)

API view - [azure-communication-common](https://apiview.dev/Assemblies/Review/c8145aaf77504830b9faf0809da1018f)

### CODE REVIEW

File: sdk/communication/azure-communication-common/src/test/java/com/azure/android/communication/common/CommunicationIdentifierTests.java [hunk 1]
```diff
@@ -113,12 +211,38 @@ public void createIdentifierFromRawId()
         assertIdentifier("4:207ffef6-9444-41fb-92ab-20eacaae2768", new PhoneNumberIdentifier("207ffef6-9444-41fb-92ab-20eacaae2768"));
         assertIdentifier("4:207ffef6-9444-41fb-92ab-20eacaae2768_207ffef6-9444-41fb-92ab-20eacaae2768", new PhoneNumberIdentifier("207ffef6-9444-41fb-92ab-20eacaae2768_207ffef6-9444-41fb-92ab-20eacaae2768"));
         assertIdentifier("4:+112345556789_207ffef6-9444-41fb-92ab-20eacaae2768", new PhoneNumberIdentifier("+112345556789_207ffef6-9444-41fb-92ab-20eacaae2768"));
-        assertIdentifier("28:45ab2481-1c1c-4005-be24-0ffb879b1130", new UnknownIdentifier("28:45ab2481-1c1c-4005-be24-0ffb879b1130"));
+
+        assertIdentifier(CommunicationIdentifier.BOT_PREFIX + microsoftBotId, new MicrosoftBotIdentifier(microsoftBotId, true));
```
Comment (petrsvihlik) [inline sdk/communication/azure-communication-common/src/test/java/com/azure/android/communication/common/CommunicationIdentifierTests.java:215]: same as in ios - use hardcoded fixtures

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

**Review by petrsvihlik — APPROVED** [review]

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            return true;
+        }
+
+        if (!(that instanceof MicrosoftBotIdentifier)) {
+            return false;
+        }
+
+        MicrosoftBotIdentifier thatId = (MicrosoftBotIdentifier) that;
+
+        if (cloudEnvironment != null && !cloudEnvironment.equals(thatId.cloudEnvironment)) {
```
Comment (AikoBB) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java:101]: will update to check only for the raw id

File: sdk/communication/azure-communication-common/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## 1.2.0-beta.1 (Unreleased)
 
 ### Features Added
+- Added support for a new communication identifier `MicrosoftBotIdentifier`.
```
Comment (petrsvihlik) [inline sdk/communication/azure-communication-common/CHANGELOG.md:6]: I'd add a record under both features added and breaking changes...see my comment in the js sdk

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

**Review by ostoliarova-msft — APPROVED** [review]

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifier.java [hunk 4]
```diff
@@ -6,6 +6,21 @@
  * Common communication identifier for Communication Services
  */
 public abstract class CommunicationIdentifier {
+    static final String ACS_USER_PREFIX = "8:acs:";
+    static final String ACS_USER_GCCH_CLOUD_PREFIX = "8:gcch-acs:";
+    static final String ACS_USER_DOD_CLOUD_PREFIX = "8:dod-acs:";
+    static final String SPOOL_USER_PREFIX = "8:spool:";
```
Comment (petrsvihlik) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifier.java:12]: interesting...we don't have this in other langs

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+     * Set full id of the identifier.
+     * RawId is the encoded format for identifiers to store in databases or as stable keys in general.
+     *
+     * @param rawId full id of the identifier.
+     * @return MicrosoftBotIdentifier object itself.
+     */
+    @Override
+    public MicrosoftBotIdentifier setRawId(String rawId) {
```
Comment (petrsvihlik) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java:89]: we could possibly avoid the whole `rawIdSet` fiddling if we didn't expose the `setRawId` method publicly. do we need to do that?

**Review by petrsvihlik — APPROVED** [review]

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java (hunk 5 above)
Comment (AikoBB) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java:89]: so as the setRawId is available for the public already from the parent class of the bot identifier, so I think we can not avoid from it

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifier.java (hunk 4 above)
Comment (petrsvihlik) [reply sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationIdentifier.java:12]: never noticed that :)

**Pull request merged**